src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,126 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (newValue: Movie) => void,
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [newMovie, setNewMovie] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+  const [isVeryfied, setIsVeryfied] = useState({
+    title: false,
+    description: false,
+    imgUrl: false,
+    imdbUrl: false,
+    imdbId: false,
+  });
+
+  const isAllVeryfied = isVeryfied.title
+    && isVeryfied.description
+    && isVeryfied.imgUrl
+    && isVeryfied.imdbUrl
+    && isVeryfied.imdbId;
+
+  const customValidation = (value: string) => {
+    // eslint-disable-next-line max-len
+    const pattern = /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@,.\w_]*)#?(?:[,.!/\\\w]*))?)$/;
+
+    const result = pattern.test(value);
+
+    return result;
+  };
+
+  const changeMovieField = (field: string, newValue: string) => {
+    const nextMovie = {
+      ...newMovie,
+      [field]: newValue,
+    };
+
+    setNewMovie(nextMovie);
+    setIsVeryfied({
+      title: !!nextMovie.title,
+      description: !!nextMovie.description,
+      imgUrl: !!nextMovie.imgUrl && customValidation(nextMovie.imgUrl),
+      imdbUrl: !!nextMovie.imdbUrl && customValidation(nextMovie.imdbUrl),
+      imdbId: !!nextMovie.imdbId,
+    });
+  };
+
+  const reset = () => {
+    setNewMovie({
+      title: '',
+      description: '',
+      imgUrl: '',
+      imdbUrl: '',
+      imdbId: '',
+    });
+  };
+
+  const handeSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (isAllVeryfied) {
+      setCount(oldValue => oldValue + 1);
+      onAdd(newMovie);
+      reset();
+    }
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={handeSubmit}
+    >
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={newMovie.title}
```

Comment:
Destructure the newMovie object
```suggestion
        value={title}
```